### PR TITLE
EVM: Implement getBlockNumber

### LINKED ISSUE #881: EVM: Implement `eth_blockNumber` endpoint.
## Summary
Implement the endpoint according to the Ethereum spec.

### PR DESCRIPTION
**Pull request by orkunkilic**
# Description
Returns the last saved block from node database.

## Linked Issues
- Fixes #881 
- Related to #502

## Testing
Describe how these changes were tested. If you've added new features, have you added unit tests?

## Docs
Describe where this code is documented. If it changes a documented interface, have the docs been updated?

### CODE REVIEW

File: module-system/module-implementations/sov-evm/src/query.rs [hunk 1]
```diff
@@ -212,13 +212,21 @@ impl<C: sov_modules_api::Context> Evm<C> {
         unimplemented!("eth_sendTransaction not implemented")
     }
 
-    // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/502
     #[rpc_method(name = "blockNumber")]
     pub fn block_number(
         &self,
```
Comment (bkolad) [inline module-system/module-implementations/sov-evm/src/query.rs:207]: Let's test it in a similar way: https://github.com/Sovereign-Labs/sovereign-sdk/pull/886#discussion_r1327209827

Comment (bkolad) [reply module-system/module-implementations/sov-evm/src/query.rs:207]: I realized that we need to wait until #870 is done with this test. LGTM, we can merge this PR and write the test later (in that case let's just comment on #881).

**Review by bkolad — APPROVED** [review]

File: module-system/module-implementations/sov-evm/src/query.rs [hunk 2]
```diff
@@ -212,13 +212,21 @@ impl<C: sov_modules_api::Context> Evm<C> {
...
-        _working_set: &mut WorkingSet<C::Storage>,
+        working_set: &mut WorkingSet<C::Storage>,
     ) -> RpcResult<reth_primitives::U256> {
-        unimplemented!("eth_blockNumber not implemented")
+        info!("evm module: eth_blockNumber");
+
+        let block_number = U256::from(
+            self.blocks
+                .last(&mut working_set.accessory_state())
```
Comment (LukaszRozmej) [inline module-system/module-implementations/sov-evm/src/query.rs]: you can just use `len()-1`

Comment (orkunkilic) [reply module-system/module-implementations/sov-evm/src/query.rs]: used saturated_sub in case there is no block (low possibility but still)

**Pull request merged**